File: problems/621/solution.py
class Solution:
    def leastInterval(self, tasks: List[str], n: int) -> int:
        abc = [0] * 26
        
        for ch in tasks:
            abc[ord(ch) - ord('A')] += 1
        
        abc.sort()
        
        m = abc[25] - 1
        freq = m * n
        
        for i in range(24, -1, -1):
            if abc[i] == 0:
                break
            
            freq -= min(m, abc[i])
        
        return len(tasks) + max(0, freq)

File: problems/621/solution.go
package main
func leastInterval(tasks []byte, n int) int {
    abc := make([]int, 26)
    
    for _, ch := range tasks {
        abc[ch - 'A']++
    }
    
    sort.Ints(abc)
    
    m := abc[25] - 1
    freq := m * n
    
    for i := 24; i >= 0; i-- {
        if abc[i] == 0 {
            break
        }
        
        freq -= min(m, abc[i])
    }
    
    return len(tasks) + max(0, freq)
}

func min(a, b int) int {
    if a < b { return a }
    return b
}

func max(a, b int) int {
    if a > b { return a }
    return b
}